Apromore-Database/src/main/java/org/apromore/dao/model/DataTypeDao.java

```diff
@@ -0,0 +1,48 @@
+/*-
+ * #%L
+ * This file is part of "Apromore Enterprise Edition".
+ * %%
+ * Copyright (C) 2019 - 2022 Apromore Pty Ltd. All Rights Reserved.
+ * %%
+ * NOTICE:  All information contained herein is, and remains the
+ * property of Apromore Pty Ltd and its suppliers, if any.
+ * The intellectual and technical concepts contained herein are
+ * proprietary to Apromore Pty Ltd and its suppliers and may
+ * be covered by U.S. and Foreign Patents, patents in process,
+ * and are protected by trade secret or copyright law.
+ * Dissemination of this information or reproduction of this
+ * material is strictly forbidden unless prior written permission
+ * is obtained from Apromore Pty Ltd.
+ * #L%
+ */
+
+package org.apromore.dao.model;
+
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.EnumType;
+import javax.persistence.Enumerated;
+import javax.persistence.Table;
+import javax.persistence.UniqueConstraint;
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import org.springframework.beans.factory.annotation.Configurable;
+
+@Entity
+@Table(name = "data_type",
+    uniqueConstraints = {
+        @UniqueConstraint(columnNames = {"id"})
+    }
+)
+@Configurable("data_type")
+@Getter
+@Builder
+@NoArgsConstructor
+@AllArgsConstructor
+public class DataTypeDao extends BaseEntity {
+    @Column(name = "type")
```

Comment:
It's a bit confusing when only particular entity is named <EntityName>Dao while others are not. DataType could be renamed DataTypeEnum and keeps this one DataType for entities like the others.

Reply:
Yeah. I went for this since the Enum is what will be used across the other modules when setting types, and not so much the dao. The dao would only be used by the persistence layer when mapping between the enum to the dao. Thought the code will read better as DataType instead of DataTypeEnum.

Reply:
Another suggestion is: TypeEnum is shorter and relevant and leave DataType for entity. If you read the code here, you're a little bit confused with DataTypeDao already :).

Reply:
The only reason you see those two together at the moment is because of the the Apromore EE / Core constraint. As mentioned the other day, I will be moving the enums into an seperate package within the process log definition module, and the dao entities into a seperate model package. but this work depends on moving Apromore database from the open source repo to the closed source repo.

Couldn't really use TypeEnum cause we have AttributeType and DataType

Reply:
Also another reason I avoided suffixing an enum with `Enum` at the end was based on this https://stackoverflow.com/questions/3069743/coding-conventions-naming-enums